app/src/androidTest/kotlin/org/wordpress/aztec/demo/tests/ImageTests.kt

```diff
@@ -24,8 +24,28 @@ class ImageTests : BaseTest() {
     @JvmField
     val mActivityIntentsTestRule = IntentsTestRule<MainActivity>(MainActivity::class.java)
 
+    // Simple photo test, also tests the issue described in
+    // https://github.com/wordpress-mobile/AztecEditor-Android/issues/306
     @Test
     fun testAddPhoto() {
+        val regex = Regex("<img src=.+>")
+
+        createImageIntentFilter()
+
+        EditorPage()
+                .tapTop()
+                .insertMedia()
+
+        // Must wait for simulated upload
+        Thread.sleep(10000)
```

Comment:
10s is quite a large delay. There no way to dynamically check if the upload is done?

Reply:
I was using the same general approach in the previously added image tests with that sleep, but I can take another look to see what options there might be to dynamically check the upload status.

Reply:
The [simulated upload](https://github.com/wordpress-mobile/AztecEditor-Android/blob/b72568f0a087ad94a09629b57039f54068a87f8d/app/src/main/kotlin/org/wordpress/aztec/demo/MainActivity.kt#L288) in the demo app takes 10s, so I'm going to leave this as-is for our Aztec tests. I'll look at a more dynamic approach for the WordPress app tests.